### PR TITLE
Switch to Gradle

### PR DESCRIPTION
**Pull request by BigBadE**
This is a full swap to gradle using examples from https://github.com/aikar/util/blob/master/build.gradle and https://github.com/aikar/db/blob/master/build.gradle.
Allows a start on #2 
Should have no deprecated features, and I generated the wrappers for Gradle 6.5

### CODE REVIEW

File: core/src/main/java/co/aikar/taskchain/TaskChain.java [hunk 1]
```diff
@@ -35,12 +35,7 @@
 
 import co.aikar.taskchain.TaskChainTasks.*;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
+import java.util.*;
```
Comment (Joo200) [inline core/src/main/java/co/aikar/taskchain/TaskChain.java]: Why did you changed this file?

I don't like import wildcards. Maybe you can revert the changes from this file.
https://stackoverflow.com/a/147461

File: core/src/main/java/co/aikar/taskchain/TaskChain.java [hunk 2]
```diff
@@ -183,7 +177,7 @@ public static void abort() {
     /**
      * Allows you to call a callback to insert tasks into the chain without having to break the fluent interface
      *
-     * Example: Plugin.newChain().sync(some::task).configure(chain -> {
+     * Example: Plugin.newChain().sync(some::task).configure(chain -&gt; {
```
Comment (Joo200) [inline core/src/main/java/co/aikar/taskchain/TaskChain.java:193]: What happened to "->" here?

Comment (BigBadE) [reply core/src/main/java/co/aikar/taskchain/TaskChain.java:193]: Javadocs are in HTML, so > is a restricted character. &gt; is the replacement for > in HTML.

File: core/src/main/java/co/aikar/taskchain/TaskChain.java (hunk 1 above)
Comment (BigBadE) [reply core/src/main/java/co/aikar/taskchain/TaskChain.java]: Because it went against the specified code style. I updated the code in my latest commit

File: .idea/jarRepositories.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      <option name="name" value="central" />
+      <option name="url" value="https://repo1.maven.org/maven2" />
+    </remote-repository>
+    <remote-repository>
+      <option name="id" value="sponge-maven-repo" />
+      <option name="name" value="Sponge maven repo" />
+      <option name="url" value="http://repo.spongepowered.org/maven" />
+    </remote-repository>
+    <remote-repository>
+      <option name="id" value="github" />
```
Comment (aikar) [inline .idea/jarRepositories.xml]: ???

File: .idea/jarRepositories.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aikar) [inline .idea/jarRepositories.xml]: Lets drop this file.

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ *  LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ *  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ *  WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+plugins {
```
Comment (aikar) [inline build.gradle]: This project doesn't use Lombok.

File: .gitignore [hunk 6]
```diff
@@ -27,3 +28,8 @@ hs_err_pid*
 .classpath
 .project
 
+# Gradle Files
+.gradle
```
Comment (aikar) [inline .gitignore]: `*/build`

File: .idea/jarRepositories.xml (hunk 3 above)
Comment (BigBadE) [reply .idea/jarRepositories.xml]: I have no clue, I usually just exclude the whole .idea file, guess some new files snuck in.

File: .idea/compiler.xml [hunk 7]
```diff
@@ -1,29 +1,16 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (aikar) [inline .idea/compiler.xml]: we can drop this file and add to ignore.

File: .idea/modules.xml [hunk 8]
```diff
@@ -6,6 +6,10 @@
       <module fileurl="file://$PROJECT_DIR$/.idea/core.iml" filepath="$PROJECT_DIR$/.idea/core.iml" />
       <module fileurl="file://$PROJECT_DIR$/.idea/sponge.iml" filepath="$PROJECT_DIR$/.idea/sponge.iml" />
       <module fileurl="file://$PROJECT_DIR$/.idea/taskchain.iml" filepath="$PROJECT_DIR$/.idea/taskchain.iml" />
+      <module fileurl="file://$PROJECT_DIR$/bukkit/taskchain-bukkit.iml" filepath="$PROJECT_DIR$/bukkit/taskchain-bukkit.iml" />
```
Comment (aikar) [inline .idea/modules.xml]: dupe modules here but already have the .iml files on ignore so i think we just need to drop this file and ignore it.

File: .idea/jarRepositories.xml (hunk 4 above)
Comment (BigBadE) [reply .idea/jarRepositories.xml]: Fixed now

**Review by aikar — CHANGES_REQUESTED** [review]

**Comment by BigBadE** [comment]
Removed lombok, and removed the poms now. The switch is complete.

**Comment by lewysDavies** [comment]
Can we keep Maven support? Or is this project going full Gradle :/ ?

**Comment by MiniDigger** [comment]
The idea is to move to gradle so that a forge platform can be implemented iirc.

you as an end user still can use maven to pull in the dependency.

**Comment by BigBadE** [comment]
What is the status on this being merged? I believe all requested changes are met.

And yes, the idea is so it can work with forge.

**Comment by BigBadE** [comment]
Bumping

**Comment by BigBadE** [comment]
Closed due to age without a maintainer's response. Someone else can work on this if they want.
